docs/src/chapters/testy.adoc

```diff
@@ -1,12 +1,262 @@
+klasy z testami integracyjnymi korzystają z klasy _BaseIntegrationTest_.
+
+Na poniższym zrzucie ekranu został załączony raport dotyczący pokrycia testami kodu związanego z częścią serwerową
+aplikacji. Wynik na poziomie 91% świadczy o solidnym podejściu do testowania stworzonego kodu. Niektóre klasy zostały
+pokryte testami nawet w 100%, co wymagało poświęcenia trochę dodatkowego czasu.
+
+.Raport z pokrycia testami kodu części backendowej
+image::../images/tests_chapter/test_coverage_backend.png[]
+
+==== Frontend
+
+Testy dotyczące części frontendowej aplikacji zostały umieszczone w katalogu _tests_. Znajdujące się w nim foldery
+zostały przedstawione na poniższym zrzucie ekranu, są to:
+
+.Folder zawierający testy do części frontendowej aplikacji
+image::../images/tests_chapter/tests_folders_frontend.png[]
+
+* components- Znalazły się tutaj testy sprawdzające, czy poszczególne elementy renderują się poprawnie
+
+.Folder zawierający testy komponentów
+image::../images/tests_chapter/components_tests.png[]
+
+* concern- W folderze umieszczony został plik zawierający atrapę historii przeglądarki.
+
+.Folder zawierający atrapę historii
+image::../images/tests_chapter/concern_folder.png[]
+
+* mockData- Jak sama nazwa wskazuje, zostały tutaj utworzone pliki z atrapami dla m.in. płatności oraz transakcji.
+
+.Folder zawierający atrapy
+image::../images/tests_chapter/mockdata_folder.png[]
+
+* pages- Znalazły się tutaj testy, w których sprawdzano np. czy po podaniu odpowiednich danych w formularzu
+administrator zostanie prawidło utworzony.
+
+.Folder zawierający testy różnych funkcjonalności
+image::../images/tests_chapter/pages_folder.png[]
+
+* utils- Katalog ten zawiera testy sprawdzające datę oraz czas
+
+.Folder zawierający testy związane z czasem
+image::../images/tests_chapter/utils_folder.png[]
+
+Fragment kodu umieszczony niżej przedstawia test odpowiedzialny za sprawdzanie, czy komponent _navbar_ zostanie
+wygenerowany w poprawny sposób.
+
+.Test generujący komponent z pliku Navbar.test.java
+[source, javascript]
+----
+include::../../../webapp/tests/components/common/Navbar.test.tsx[lines=6..12]
+----
+
+Kolejny test zaprezentowany poniżej odpowiedzialny jest za sprawdzenie formularza zmiany hasła. Po wprowadzeniu
+prawidłowych danych oczekiwany jest komunikat o powodzeniu operacji.
+
+.Test sprawdzający formularz zmiany hasła z pliku PasswordChangePage.test.java
+[source, javascript]
+----
+include::../../../webapp/tests/pages/userPanel/account/PasswordChangePage.test.tsx[lines=25..42]
+----
+
+Kod napisany w _Typescript_ został pokryty testami w ponad 65%. Stworzono 32 przypadki testowe, zawierające 93 testy.
+Jest to rezultat widocznie gorszy niż w przypadku części serwerowej. Pewien wpływ miały na to zapewne opóźnienia w pracy
+nad projektem i brak dodatkowego czasu na implementację kolejnych testów.
+
+.Raport z pokrycia testami kodu części frontendowej
```

Comment:
To samo co wcześniej, co z raportem z jest-test-reporter?

Reply:
to jest pokrycie kodu, które pokazuje się po odpaleniu testów przez jesta (komenda test:unit). W typ raporcie html jest tylko ilość przypadków testowych,  testów  i wypisane poszczególne testy, które zostały wykonane. Według mnie w takim raporcie powinny być dane % jak mam w inny sposób to wygenerować to napisz czego oczekujesz i ogarnę.

Reply:
dobre